public_html/storefront/controller/pages/account/create.php

```diff
@@ -370,6 +370,8 @@ public function main()
         $text_account_already = sprintf($this->language->get('text_account_already'), $this->html->getSecureURL('account/login'));
         $this->data['text_account_already'] = $text_account_already;
 
+        $this->extensions->hk_ProcessData($this,'account_create');
```

Comment:
Why you added this?
You can change data via $this->view inside your hook.
For example:
$that = $this->baseObject;
$viewData = $that->view->getData();    // this will be this->data from controller

Then you can add your custom data directly
$that->view->assign('some_var_name', 'some_value');

I think no need ProcessData here.

Reply:
Suppose i want to remove the $this->data['form']['fields']['general'['telephone']. How can i do that by using the existing hook?

Reply:
public function onControllerPagesAccountEdit_UpdateData()
{
$that = $this->baseObject;
$form = $that->view->getData('form');
unset( $form['fields']['telephone']);

$that->view->assign('form', $form); //probably this line don't needed here, you should check (as i remember $form var will be a reference, but i'm not sure)
}


As you see you don't need an additional hooks-call here. You can do this inside "UpdateData" before template rendering.

Please remove your ProcessData calls and i'll merge your hookvars into 1.3.0 branch.

Reply:
I have remove the ProcessData. Please check them again.

Reply:
I just test your solution. The `$that->view->assign('form', $form);` is needed here. I am able to remove the unwanted fields now.

Thanks a lot.